.github/workflows/test.yml

```diff
@@ -21,4 +21,4 @@ jobs:
     uses: ./.github/workflows/test_template.yml
     with:
       runs-on: '["ubuntu-latest", "macos-latest", "windows-latest"]'
-      python-version: '["3.8", "3.9", "3.10", 3.11]'
+      python-version: '["3.9", "3.10", 3.11]'
```

Comment:
> I guess 3.12 should be added here? I see that there is a 3.12-pre in the pre workflow, but since 3.8 is dropped, and given that 3.12 is out (and assuming dependencies are already compatible), it may be wise to add it here and transition the pre to 3.13 (if applicable)?

Unfortunately, h5py does not have wheels for 3.12 and compilation failed last time I checked. So, for now, I keep it in pre until they fix it. 

> Wow, dropping 3.8 may impact many users. Not sure if this was anticipated in the previous release, but this is likely to cause unhappiness. I know it may be difficult to plan ahead at times, but IMO communicating this sufficiently in advance would have been necessary (have not found any relevant statement in the release notes).

ok, you go a point here, the end of life is in 1 year, we can keep 6 month more... I forgot the reason of the drop, I need to check again.

Thank you for the feedback !

Reply:
I believe most of the core packages have already dropped 3.8 and are about to drop 3.9:
https://scientific-python.org/specs/spec-0000/

Reply:
Thank you for sharing @jarrodmillman,

So I will keep it as it.

Also, I will read all the documents, and I think we will start following the scientific-python standard and eco-system.  is there anything to do to be part of this scientific-python ecosystem?

Reply:
DIPY is part of the ecosystem, of course. But following and contributing to development of the SPECs would be great. We are also looking into mechanisms to let projects indicate when they've implemented a SPEC.

Reply:
Either way, and although a mainenance branch can be opened to backport/aply some critical fixes, IMO users need to be informed beforehand about changrs of this caliber. If we stick to the scientific python conventions, then this needs to be made explicit so that users are aware of it from now on.

Given that this time they were not warned, and unless keeping 3.8 makes things a lot more complicated, I think we should keep it for this release cycle/until its EOL next year.